dbms/src/Dictionaries/DictionaryStructure.cpp

```diff
@@ -74,6 +74,11 @@ AttributeUnderlyingType getAttributeUnderlyingType(const std::string & type)
             return AttributeUnderlyingType::utDecimal128;
     }
 
+    // Temporary hack to allow arrays in keys, since they are never retrieved for polygon dictionaries.
+    // TODO: This should be fixed by fully supporting arrays in dictionaries.
+    if (type.find("Array") == 0)
```

Comment:
one nice addiction would be to allow String keys in WKB format (which is pretty easy to generate with different tools or export from postgis)

GeoJSON, WKT and other formats would be nice to have.

Happy to contribute on this when the PR is finished and you think it's feasible.

Reply:
This makes a lot of sense, but we decided that the first priority right now is implementing and testing different approaches for efficient lookup. 
If you were to contribute that'd be great! Otherwise, we will hopefully get around to this once we have an efficient implementation.
There are also some students working, albeit slowly, on implementing geometric primitive types. Maybe they will cross this as well.

Reply:
I will keep an eye on this PR and will try to contribute (looks like boost already have a parser so looks "easy")

thanks for this, pretty much needed.